test/js/client.js

```diff
@@ -0,0 +1,129 @@
+
+      it("works with default expose", function() {
+        process.env.SERVICE_SWAGGER_TEST_DEFAULT_PROTO = "http"
+        process.env.SERVICE_SWAGGER_TEST_DEFAULT_HOST = "localhost"
+        process.env.SERVICE_SWAGGER_TEST_DEFAULT_PORT = "8000"
+        c = new Client({discovery: true});
+        assert.equal(c.address, mockAddress);
+      });
+
+      it("works with legacy `http` expose", function() {
+        process.env.SERVICE_SWAGGER_TEST_HTTP_PROTO = "http"
+        process.env.SERVICE_SWAGGER_TEST_HTTP_HOST = "localhost"
+        process.env.SERVICE_SWAGGER_TEST_HTTP_PORT = "8000"
+        c = new Client({discovery: true});
+        assert.equal(c.address, mockAddress);
+      });
+    });
+  });
+
+  describe("GET", function() {
+    it("correctly fills out path, header and query params", function(done) {
+      const c = new Client({address: mockAddress});
+      const params = {
+        authorization: "Bearer YOGI",
+        bookID: "the-mediocre-gatsby",
+        randomBytes: "4" // Very random
+      }
+      const mockResponse = {hello: "world"};
+      const scope = nock(mockAddress, {reqheaders: {'authorization': params.authorization}})
+        .get(`/v1/books/${params.bookID}`)
+        .query({randomBytes: params.randomBytes})
+        .reply(200, mockResponse);
+      c.getBookByID(params, function(err, resp) {
+        assert(scope.isDone());
+        assert.ifError(err);
+        assert.deepEqual(resp, mockResponse)
+        done();
+      });
+    });
+
+    it("accepts an options argument", function(done) {
+      const c = new Client({address: mockAddress});
+      const scope = nock(mockAddress)
+        .get(`/v1/books`)
+        .reply(200);
+      c.getBooks({}, {}, function() {
+        assert(scope.isDone());
+        done();
+      });
+    });
+
+    it("returns a error in failure cases", function(done) {
+      const c = new Client({address: mockAddress});
+      const scope = nock(mockAddress)
+        .get(`/v1/books`)
+        .reply(400, {problems: 99});
+      c.getBooks({}, {}, function(err, resp) {
+        assert.equal(resp, undefined);
+        assert.equal(err.message, "[object Object]"); // TODO
+        done();
+      });
+    });
+
+    it("returns a promise", function(done) {
+      const c = new Client({address: mockAddress});
+      const scope = nock(mockAddress)
+        .get(`/v1/books`)
+        .reply(200);
+      c.getBooks({}, {}).then(function() {
+        assert(scope.isDone());
+        done();
+      });
+    });
+
+    it("returns a error to the promise", function(done) {
+      const c = new Client({address: mockAddress});
+      const scope = nock(mockAddress)
+        .get(`/v1/books`)
+        .reply(400, {problems: 99});
+      c.getBooks({}, {}).then(function() {
+        assert(false, "then callback should not have been called");
+      }).catch(function(err) {
+        assert.equal(err.message, "[object Object]"); // TODO
+        done();
+      });
+    });
+  });
+});
```

Comment:
naming this file `client.js` seems like a missed opportunity... as is it communicates zero information (all the tests in this directory are client tests). Can you think of a logical grouping for the tests in this file?